devel/irrlichtmt/Portfile

```diff
@@ -2,20 +2,178 @@
 
 PortSystem          1.0
 PortGroup           github 1.0
-PortGroup           cmake 1.1
+PortGroup           cmake  1.1
+
+# NOTE: See the comprised 'subport' definitions of this Portfile below.
+#       'irrlichtmt' referred by the applicable 'subport' from the 'minetest' Portfile.
 
-github.setup        minetest irrlicht 1.9.0mt8
 name                irrlichtmt
+
+set github-author   minetest
+set github-project  irrlicht
+
+# Amends to body part of both the port and the most current subport only.
+if { ${subport} eq ${name} || ${subport} eq "${name}-mt57" } {
+
+# Set this to the most current version of the Minetest fork of IrrLicht as applicable.
+    github.setup     ${github-author} ${github-project} 1.9.0mt9
+}
+
+revision            0
+
+# MT core version stub for the quite "unused" main 'irrlichtmt' port envelope.
+set mtbuild         unused
+
+homepage            https://github.com/minetest/irrlicht
+
 categories          devel
 license             zlib
-maintainers         {l2dy @l2dy} openmaintainer
+maintainers         {l2dy @l2dy} @Zweihorn openmaintainer
 description         a fork of Irrlicht by the Minetest developers
-long_description    IrrlichtMt is ${description}.
 
-checksums           rmd160  4da410c24f67e8c0ebd9880308b8d796d2f86676 \
+# Checksums for the current irrlichtmt
+checksums           rmd160  9b752b255cea2c443474f78aeadba244e8f584a2 \
+                    sha256  a81b4c362d724ce4bdb2c8f60e31df0b1bff10302e7e0458248f9d609bd12f16 \
+                    size    908173
+
+
+# NOTE: There might be the argument to avoid conflicts if this might be possible.
+#       However, this seems to be a more complex task and might not easily feasible.
+# Rationale:  The Minetest core evidently binds to a specific IrrLichtMT fork version.
+# The co-maintainer is not aware of any reasonable use case for different MT core versions
+# being required as parallel install.  To the opposite this might confuse MT players greatly.
+# Furthermore, a subport can be easily activated and deactivated without install like any port.
```

Comment:
Do you expect users to manually deactivate/uninstall an older version of irrlichtmt before installing a new version of minetest? This would make the upgrade path more difficult.

Reply:
mintest ports don't have to be co-installable, but library ports should be to ease upgrading.

Reply:
> Do you expect users to manually deactivate/uninstall an older version of irrlichtmt before installing a new version of minetest? This would make the upgrade path more difficult.

There is need for any uninstall, a simple 'deactivate' will suffice, I presume.

@l2dy - What is your rationale for the a.m. claim "would make the upgrade path more difficult" pleas?

Reply:
> mintest ports don't have to be co-installable, but library ports should be to ease upgrading.

@l2dy - THX for resolving on my general approach with 'minetest' and I will proceed #17258 accordingly.

Don't want to bother, but please clarify more on a.m. claim "but library ports should be to ease upgrading."
Certainly this was no feature found with the 'irrlichtmt' port before my inventions, I presume.

Furthermore, please have a look at
- https://trac.macports.org/wiki/howto/InstallingOlderPort

IMO my current approach with `irrlichtmt` & subports seems to quite consistent with the procedure described in the a.m. trac guide article.

Please prove me wrong and in addition provide an applicable technical solution for a both concurrent & activated of several versions of one port or at least some documentary advice to be found at MacPorts on this.

Nevertheless, I appreciate the opportunity for learning from this discussion. Last not least, I highly appreciate your commitment and you gain investing some more time with a new co-maintainer and an imbecile like myself in particular.

BTW - **Clarification:** My former wording of _"parallel" install_ might have been ambiguous or even misleading. The more applicable wording should be **"concurrently" install*ed*, I presume.

> concurrently | kənˈkʌrəntli |
> adverb
> at the same time; simultaneously

Please accept my apologies and English is a foreign language to me.

Hope this helps.

🌻

Reply:
> > mintest ports don't have to be co-installable, but library ports should be to ease upgrading.
> 
> @l2dy - THX for resolving on my general approach with 'minetest' and I will proceed #17258 accordingly.
> 
> Don't want to bother, but please clarify more on a.m. claim "but library ports should be to ease upgrading." Certainly this was no feature found with the 'irrlichtmt' port before my inventions, I presume.

Before this change, users could install `minetest` normally and get updates with `sudo port upgrade outdated`.

After your changes, to install a newer version of minetest, the user has to deactivate (or uninstall) both minetest and irrlichtmt ports first, which is less intuitive to not so tech-savvy users. `port upgrade outdated` would not work anymore, so every upgrade requires manually intervention.

Burden on the port maintainers is also increased:

1. How many versions of irrlichtmt do you intend to keep? If unbounded, the `conflicts` lines in Portfile would become longer and longer.
2. Assume you are a user who only knows how to install and uninstall ports, what would you do when seeing this?

```
Error: Can't install minetest55 because conflicting ports are active: minetest54
Error: Follow https://guide.macports.org/#project.tickets if you believe there is a bug.
Error: Processing of port minetest55 failed
```

The user is not given instructions on how to resolve this, and may just submit a ticket thinking it could be a bug, further increasing maintenance burden.

> Furthermore, please have a look at
> 
>     * https://trac.macports.org/wiki/howto/InstallingOlderPort
> 
> 
> IMO my current approach with `irrlichtmt` & subports seems to quite consistent with the procedure described in the a.m. trac guide article.

This wiki page is an advanced guide about installing an older version of the same port. It is "not usually recommended" and I would assume less than 1% of MacPorts users have tried it.

> Please prove me wrong and in addition provide an applicable technical solution for a both _concurrently & activated_ install of several versions of one port or at least some documentary with valid advice to be found at MacPorts on this particularity.

There are several examples in this repository, for example the `openssl` ports. Do your own research about this.

> Nevertheless, I appreciate the opportunity for learning from this technical discussion. Last not least, I highly appreciate your personal commitment and you again investing some more time with a new co-maintainer and an imbecile like myself in particular.
> 
> BTW - **Clarification:** My former wording of _"parallel" install_ might have been ambiguous or even misleading. A more applicable wording should be _**"concurrently"** installed_, I presume.
> 
> > concurrently | kənˈkʌrəntli |
> > adverb
> > at the same time; simultaneously
> 
> Please accept my apologies and English is a foreign language to me.

Parallel is the right word to use, don't worry about that. You should be specific about who you are addressing in the `NOTE` comment though. In this context, co-maintainer could mean either of us.

On the other hand, be careful with your tone. In the open source world, "free and open endeavour" does not grant you entitlement to maintainers' attention, nor does it guarantee that your PR would be merged. Maintainers have a decision to make, and it's not always the best choice to add more features.

Lastly, I do not feel comfortable adding you as a co-maintainer until you get more familiar with MacPorts mechanics and conventions, so please revert the `maintainers` change for now.

Reply:
> On the other hand, be careful with your tone. In the open source world, "free and open endeavour" does not grant you entitlement to maintainers' attention, nor does it guarantee that your PR would be merged. Maintainers have a decision to make, and it's not always the best choice to add more features.

That is quite funny. However, I might not get your undertone correctly, I presume.

Reply:
> Lastly, I do not feel comfortable adding you as a co-maintainer until you get more familiar with MacPorts mechanics and conventions, so please revert the maintainers change for now.

That does not surprise me very much. I stepping in your garden obviously and you want to protect your flowers.

I am quite neutral on the co-maintainer and not my cup of tea anyway. I am not claiming the code and don't want to become king of MacPorts in two days time. However, maybe three?

Just joking. Your deliberation.

Have fun.

🌻

Reply:
> After your changes, to install a newer version of minetest, the user has to deactivate (or uninstall) both minetest and irrlichtmt ports first, which is less intuitive to not so tech-savvy users. port upgrade outdated would not work anymore, so every upgrade requires manually intervention.

There seems to be misunderstanding. Please clarify.

Obviously, the `port upgrade outdated` would function for each subport.

Reply:
IMO you are mixing some issues of the draft `minetest` with the more mature `irrlichtmt` subport approach.

However, I will consider the technical aspects of your reasoning indue time.

THX

🌻

Reply:
This PR was made obsolete by #17419